src/main/java/org/kuali/kfs/module/purap/document/web/struts/PurchasingActionBase.java

```diff
@@ -1647,9 +1655,11 @@ private void createReasonNote(ActionForm form) {
     	KualiDocumentFormBase kualiDocumentFormBase = (KualiDocumentFormBase) form;
         try {
             Note noteObj = SpringContext.getBean(DocumentService.class).createNoteFromDocument(kualiDocumentFormBase.getDocument(), ((PurchasingFormBase)kualiDocumentFormBase).getReasonToChange());
+            populateIdentifierOnNoteAndExtension(noteObj);
             kualiDocumentFormBase.getDocument().addNote(noteObj);
-         //   SpringContext.getBean(NoteService.class).save(noteObj);
-            SpringContext.getBean(NoteService.class).saveNoteList(kualiDocumentFormBase.getDocument().getNotes());
+            if (isNoteReadyToBeSaved(kualiDocumentFormBase.getDocument(), noteObj)) {

```

Comment:
The reason-note code is available to all PURAP documents, but according to our note tag customizations, only REQS and POs are actively using it.

The isNoteReadyToBeSaved method is trying to incorporate some of the should-save-note checks from the base insertBONote method for document actions. Of those checks, though, I think only the is-not-INITIATED-status check is the only one that would be applicable here (since this action class is for transactional docs only, and such docs typically have a note target already).

If there's cases where certain other PO-related docs (like POA) can change the "Send to Vendor" flag on the base PO doc's notes while in INITIATED status, then we may need to keep at least the is-not-INITIATED-status check. Otherwise, it should be safe to remove the isNoteReadyToBeSaved method altogether.